app/channel.go

```diff
@@ -27,6 +27,7 @@ import (
 // channelsWrapper provides an implementation of `product.ChannelService` to be used by products.
 type channelsWrapper struct {
 	srv *Server
+	app *App
```

Comment:
This seems incorrect, we are using both the server and the app. Whereas, the server is accessible from the app. If we need app methods, we should just use app only.

cc @isacikgoz

Reply:
Added my comment above, we can use app and access server from there if necessary. Or always use the app.

Reply:
Yep, let's always use the app and access server only if necessary.